pkg/fixer/fixes/nowhitespacecomment_test.go

```diff
@@ -0,0 +1,120 @@
+package fixes
+
+import (
+	"testing"
+
+	"github.com/open-policy-agent/opa/ast"
+)
+
+func TestNoWhitespaceComment(t *testing.T) {
+	t.Parallel()
+
+	testCases := map[string]struct {
+		runtimeOptions *RuntimeOptions
+
+		beforeFix []byte
+		afterFix  []byte
+
+		fixExpected bool
+	}{
+		"no change needed": {
+			beforeFix: []byte(`package test\n
+
+# this is a comment
+`),
+			afterFix: []byte(`package test\n
+
+# this is a comment
+`),
+			fixExpected:    false,
+			runtimeOptions: &RuntimeOptions{},
+		},
+		"no change made because no location": {
+			beforeFix: []byte(`package test\n
+
+#this is a comment
+`),
+			afterFix: []byte(`package test\n
+
+#this is a comment
+`),
+			fixExpected:    false,
+			runtimeOptions: &RuntimeOptions{},
+		},
+		"single change": {
+			beforeFix: []byte(`package test\n
+
+#this is a comment
+`),
+			afterFix: []byte(`package test\n
+
+# this is a comment
+`),
+			fixExpected: true,
+			runtimeOptions: &RuntimeOptions{
+				Locations: []ast.Location{
+					{
+						Row: 3,
+						Col: 1, // this is what the rule outputs at the moment
```

Comment:
Is it? AFAICS I get the col where the comment starts
```
Rule:         	no-whitespace-comment
Description:  	Comment should start with whitespace
Category:     	style
Location:     	p.rego:13:3
Text:         	} #doo
Documentation:	https://docs.styra.com/regal/rules/style/no-whitespace-comment
```

Reply:
Huh, I could have sworn this was only for the first char... fixed in fa62e1865e9484ca50014b86833b01b845fefa61.

I think that opa fmt must have trimmed it on save and it was moved to the first char in my test file...